### PR TITLE
Pretokenize

### PR DESCRIPTION
**Pull request by np179**

### CODE REVIEW

File: src/annotator/test/test_files/test.vrt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<s>
+This	DT	this	-	nsubj	PRON
```
Comment (iulusoy) [inline src/annotator/test/test_files/test.vrt:2]: This is not a valid vrt, though, right? Due to the "-", where does it come from?

File: src/annotator/test/test_msomajo.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+@pytest.fixture
+def read_test_en():
+    with open("test/test_files/example_en_somajo.vrt", "r") as f:
+        data = f.read()
+    return data
+
+
+def test_tokenize(read_test_de, read_test_en):
+
+    text_de = "Das Feld der Computerchemie erfreut sich seit einiger Zeit einer stetig zunehmenden Aufmerksamkeit. Hierbei ist es für die Berechnung von chemisch relevanten Systemen von Interesse, möglichst große Systeme (Moleküle) durch effiziente Nutzung von Rechen- und Speicherressourcen zugänglich zu machen. Die Herausforderung besteht darin, dass mit der zunehmenden Anzahl von Atomen in größeren Systemen auch die Anzahl der internen Freiheitsgrade zunimmt."
```
Comment (iulusoy) [inline src/annotator/test/test_msomajo.py:21]: The tests are saved in `test_files/example_en.txt` and `test_files/example_de.txt`, for keeping it general should read from there.

File: src/annotator/test/test_mtreetagger.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+@pytest.fixture
+def read_test_de():
+    with open("test/test_files/example_de_treetagger.vrt", "r") as f:
+        data = f.read()
+    return data
+
+
+def test_tokenize(read_test_de, read_test_en):
+
+    text_de = "Das Feld der Computerchemie erfreut sich seit einiger Zeit einer stetig zunehmenden Aufmerksamkeit. Hierbei ist es für die Berechnung von chemisch relevanten Systemen von Interesse, möglichst große Systeme (Moleküle) durch effiziente Nutzung von Rechen- und Speicherressourcen zugänglich zu machen. Die Herausforderung besteht darin, dass mit der zunehmenden Anzahl von Atomen in größeren Systemen auch die Anzahl der internen Freiheitsgrade zunimmt."
```
Comment (iulusoy) [inline src/annotator/test/test_mtreetagger.py:21]: Same here, should read the texts from the provided files as in `test_mstanza.py` etc.

File: src/annotator/test_cwb_encoding_decoding.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+if __name__ == "__main__":
+    data = be.prepare_run.get_text("src/annotator/test/test_files/example_de.txt")
+    # or read the main dict and activate
+    mydict = be.prepare_run.load_input_dict("src/annotator/input_local")
+    # take only the part of dict pertaining to spacy
+    # filename needs to be moved to/taken from top level of dict
+    # spacy_dict = mydict["spacy_dict"]
+    # remove comment lines starting with "_"
+    # for now, we are not using "components" as these are defined through the pre-
```
Comment (iulusoy) [inline src/annotator/test_cwb_encoding_decoding.py:13]: These could also become part of the integration tests mspacy and mstanza.

File: src/annotator/test_somajo.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import annotator.msomajo as msomajo
+import annotator.base as be
+import annotator.mtreetagger as mtreetagger
+
+text = "Das Feld der Computerchemie erfreut sich seit einiger Zeit einer stetig zunehmenden Aufmerksamkeit. Hierbei ist es für die Berechnung von chemisch relevanten Systemen von Interesse, möglichst große Systeme (Moleküle) durch effiziente Nutzung von Rechen- und Speicherressourcen zugänglich zu machen. Die Herausforderung besteht darin, dass mit der zunehmenden Anzahl von Atomen in größeren Systemen auch die Anzahl der internen Freiheitsgrade zunimmt."
```
Comment (iulusoy) [inline src/annotator/test_somajo.py:5]: Read from file.

File: src/annotator/test_somajo.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+if __name__ == "__main__":
```
Comment (iulusoy) [inline src/annotator/test_somajo.py:7]: This in main as `test_integration_msomajo.py` (see spacy/stanza).

**Review by iulusoy — COMMENTED** [review]
Added a few comments, minor things, please take a look.

File: src/annotator/test/test_files/test.vrt (hunk 1 above)
Comment (np179) [reply src/annotator/test/test_files/test.vrt:2]: If I'm not mistaken this is due to the fact that the corresponding Tag has not been defined. I think we need something as a placeholder though right? The "-" is defined in the base.out_object.grab_ functions. I can change it to a whitespace or anything else.

Comment (np179) [reply src/annotator/test/test_files/test.vrt:2]: If it is a whitespace it will be encoded as __UNDEF__ in CWB

**Pull request merged**